### PR TITLE
feat(clerk-js,types): Support connecting Coinbase Wallet via <UserProfile />

### PR DESCRIPTION
**Pull request by chanioxaris**
## Description

Allow users to connect their Coinbase Wallet to their account via `<UserProfile />`

https://github.com/user-attachments/assets/6989682b-a938-4b2f-9ddb-e76e52cbd92c

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/clerk-js/src/utils/web3.ts [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
-export async function getMetamaskIdentifier(): Promise<string> {
+type GetWeb3IdentifierParams = {
+  provider: Web3Provider;
+};
+
+export async function getWeb3Identifier(_: GetWeb3IdentifierParams): Promise<string> {
   // @ts-ignore
   if (!global.ethereum) {
-    // Do nothing when ethereum doesn't exist. We might revise this in the future
-    // to offer an Install Metamask prompt to our users.
```
Comment (panteliselef) [inline packages/clerk-js/src/utils/web3.ts:7]: Is this recommendation not viable ? (just curious)

File: packages/clerk-js/src/ui/components/UserProfile/Web3Form.tsx [hunk 2]
```diff
@@ -62,7 +62,7 @@ export const AddWeb3WalletActionMenu = withCardStateProvider(() => {
           onClick={() => connect(strategy)}
           isLoading={card.loadingMetadata === strategy}
           isDisabled={card.isLoading}
-          localizationKey={`Connect ${strategyToDisplayData[strategy].name} wallet`}
+          localizationKey={`Connect ${strategyToDisplayData[strategy].name}`}
```
Comment (panteliselef) [inline packages/clerk-js/src/ui/components/UserProfile/Web3Form.tsx:65]: I believe it's time to create a proper localization key here

**Review by panteliselef — CHANGES_REQUESTED** [review]
Minor request about the localization key

File: .changeset/small-melons-smell.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+"@clerk/clerk-js": patch
+"@clerk/types": patch
```
Comment (panteliselef) [inline .changeset/small-melons-smell.md:3]: I'd do `minor` because it affects the User interface

**Review by alexcarpenter — COMMENTED** [review]
Doesn't look like it would be included in this PR, but highlighting that the logo - name spacing is a bit off here. The logo is almost touching the name coinbase.

Screenshot 2024-08-27 at 5 42 10 PM

Can we add a UI ticket to follow up and improve here?

**Comment by chanioxaris** [comment]
> Doesn't look like it would be included in this PR, but highlighting that the logo - name spacing is a bit off here. The logo is almost touching the name coinbase.
> 
> Screenshot 2024-08-27 at 5 42 10 PM
> Can we add a UI ticket to follow up and improve here?

Sure we can do that. Which team should handle such improvements/tickets?

File: packages/clerk-js/src/utils/web3.ts (hunk 1 above)
Comment (chanioxaris) [reply packages/clerk-js/src/utils/web3.ts:7]: Not sure if it's possible or what's the recommended way (hopefully we will find out soon), but removed the comment until we have a concrete plan in place

**Comment by chanioxaris** [comment]
> Doesn't look like it would be included in this PR, but highlighting that the logo - name spacing is a bit off here. The logo is almost touching the name coinbase.
> 
> Screenshot 2024-08-27 at 5 42 10 PM
> Can we add a UI ticket to follow up and improve here?

@alexcarpenter it was an easy fix so i went ahead and fixed that as well as part of this PR. That's how it looks right now:
Screenshot 2024-08-28 at 10 19 13 AM

File: .changeset/small-melons-smell.md (hunk 3 above)
Comment (chanioxaris) [reply .changeset/small-melons-smell.md]: Updated to use a minor version

File: packages/clerk-js/src/ui/components/UserProfile/Web3Form.tsx (hunk 2 above)
Comment (chanioxaris) [reply packages/clerk-js/src/ui/components/UserProfile/Web3Form.tsx]: I went ahead and aligned this with the Social connections section. What's changed is:
1. Update the copyright of the primary button from `Web3 wallets` to `Connect wallet`
2. Introduced a new localization key `userProfile.web3WalletPage.web3WalletButtonsBlockButton` to only render the web3 provider name on the dropdown

File: .changeset/late-dots-greet.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+"@clerk/clerk-js": minor
+"@clerk/types": minor
```
Comment (panteliselef) [inline .changeset/late-dots-greet.md:3]:
```suggestion
"@clerk/localizations": minor
```

**Review by panteliselef — APPROVED** [review]
minor comment about the changelog, otherwise looks good

**Pull request merged**